### ISSUE TITLE
Linter applies on text.html.ruby

### ISSUE THREAD

**Issue by Project0** [AtomLinter/linter-ruby#26]
for some reason, the plugin tries to lint HTML Ruby ERB files.

**Comment by MartinodF** [comment]
Same here. The ruby parts of an "HTML (Ruby - ERB)" file (those between `<%` and `%>`) are detected as 'source.ruby` and I'm guessing that triggers the plugin to lint the entire file.

![linter-ruby](https://cloud.githubusercontent.com/assets/195024/9366158/f39ca4ec-46b5-11e5-920e-0650a9a95897.png)

Atom 1.0.7
linter 1.4.3
linter-ruby 1.0.2
ruby 2.2.2p95 on Arch Linux

**Comment by Project0** [comment]
```
editor = atom.workspace.getActiveTextEditor()
editor.scopeDescriptorForBufferPosition(editor.getCursorBufferPosition()).scopes
["text.html.erb", "meta.embedded.line.erb", "source.ruby", "keyword.control.ruby"]
´´´
grammar is detect as source.ruby. I guess the problem is related to language-ruby plugin. 
Btw. i fixed that by installing the plugin "language-ruby-on-rails"

**Comment by philipgiuliani** [comment]
Could you port this also to https://github.com/AtomLinter/linter-rubocop ? We have the same problem there.

**Comment by Arcanemagus** [comment]
@philipgiuliani The "fix" in #37 just adds a configuration option to disable certain file extensions.

The proper fix for this would be for whatever package is providing your Ruby language support to properly mark the Ruby code in these files as `source.embedded.ruby` instead of `source.ruby`. Once that is fixed these linters will stop triggering on the embedded code (unless they mark themselves as supporting it).

I would imagine your problem lies in [this file](https://github.com/atom/language-ruby/blob/master/grammars/html%20(ruby%20-%20erb).cson), which would be why installing `language-ruby-on-rails` might fix it (as it might have an overriding specification?).

**Comment by emilyaviva** [comment]
Installing `language-ruby-on-rails` does not fix the problem. The "fix" in #37, as @Arcanemagus pointed out above, doesn't actually address the problem. I don't want to ignore `.html.erb` files; on the contrary, they're the ones I want to lint!
<img width="450" alt="screen shot 2015-11-05 at 12 40 53 am" src="https://cloud.githubusercontent.com/assets/12450261/10963550/1cd45580-8356-11e5-8b4c-4c9d19c1cb20.png">

**Comment by Arcanemagus** [comment]
@emilyaviva I'm not quite sure what you mean there... this linter simply runs `ruby -w -c` on the file and reports the results. From what I understand at least `.erb` files need to be ran through `erb`, not `ruby`.

**Comment by emilyaviva** [comment]
No, you're right, I had misapprehended how this works. Pay me no heed.
